src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala

```diff
@@ -104,6 +105,42 @@ object GroupReadsByUmi {
     }
   }
 
+  /** Sort the records so they are in the same order as the output of [[GroupReadsByUmi]].
+    *
+    * @param iterator an iterator from which to consume records
+    * @param header the header associated with the records.
+    * @param maxInMemory the maximum number of records to keep and sort in memory if sorting is needed
+    * @param tmpDir an optional temp directory to use for temporary sorting files if needed
+    * @return an Iterator with reads from the same query grouped together
+    */
+  def groupReadsByUmiSort(iterator: Iterator[SamRecord],
+                          header: SAMFileHeader,
+                          maxInMemory: Int = MaxInMemory,
+                          tmpDir: DirPath = Io.tmpDir): Iterator[SamRecord] = {
+    val sort = new Sorter[SamRecord, GroupReadsByUmiSortKey](maxInMemory, new SamRecordCodec(header), GroupReadsByUmiSortKey.apply, tmpDir=tmpDir)
+    sort ++= iterator
+    new SelfClosingIterator(sort.iterator, () => sort.close())
+  }
+
+  private object GroupReadsByUmiSortKey {
+    /** Defines the ordering as the tuple of elements in the order defined in [[GroupReadsByUmiSortKey]]. */
+    implicit val ord: Ordering[GroupReadsByUmiSortKey] = Ordering.by(GroupReadsByUmiSortKey.unapply)
+
+    def apply(rec: SamRecord): GroupReadsByUmiSortKey = {
+      val r = ReadInfo(rec)
+      GroupReadsByUmiSortKey(refIndex=r.refIndex, start1=r.start1, start2=r.start2, strand1=r.strand1, strand2=r.strand2, library=r.library,
+        mid=rec(ConsensusTags.MolecularId), name=rec.name, secondOfPair=rec.secondOfPair)
+    }
+  }
+
+  // NB: use secondOfPair since false sorts before true
+  /** The sort key for [[GroupReadsByUmi]]. */
+  private case class GroupReadsByUmiSortKey(refIndex: Int, start1: Int, start2: Int, strand1: Boolean, strand2: Boolean, library: String,
+                                            mid: String, name: String, secondOfPair: Boolean)
```

Comment:
NB: I could have `ReadInfo` as a value here, since I couldn't figure out some scala "diverging implicits" error.  Hence, I recreated all the values.  It would have been nicer to make `ReadInfo` extend `Ordered[ReadInfo]` and just have a reference to a `ReadInfo` object here, but alas, only so many hours.